### PR TITLE
Enhancement/2128 migrate storybook

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue #2128 

## Relevant technical choices
- Added per story decorators by passing an object containing the `decorators` key as third parameter of the `add` function. Since a lot of stories are under the "Global" kind, the decorators are shared across all stories instead of being isolated. As per the [migration guide](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-support-for-duplicate-kinds), in Storybook 6, they have deprecated the ability to split a kind's (component's) stories into multiple files and will likely be removed in v7. So there is some refactoring to be done in the future.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/util/index.js [hunk 1]
```diff
@@ -620,7 +620,7 @@ export const getTopPagesReportDataDefaults = () => {
  */
 export const parseTotalUsersData = ( data ) => {
 	return {
-		totalUsers: data?.[0]?.data?.totals?.[0]?.values?.[0],
-		previousTotalUsers: data?.[0]?.data?.totals?.[1]?.values?.[0],
+		totalUsers: data?.[ 0 ]?.data?.totals?.[ 0 ]?.values?.[ 0 ],
+		previousTotalUsers: data?.[ 0 ]?.data?.totals?.[ 1 ]?.values?.[ 0 ],
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/util/index.js:624]: Did you see an error for this? I noticed that this code block in particular was failing to be caught by ESLint with the current packages we have now.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @asvinb – this looks great. Apologies for the delay in getting back to this but we have a clearer path forward with v6 now so we're ready to move this forward.

Would you please resolve the merge conflicts and refresh the PR against develop? I would imagine there will be a few new stories to update in the process but once ready, we can move quickly to get this merged to avoid ongoing conflicts 👍

File: assets/js/modules/analytics/util/index.js (hunk 1 above)
Comment (asvinb) [reply assets/js/modules/analytics/util/index.js:627]: @aaemnnosttv Yes, eslint was complaining about it :(

**Comment by asvinb** [comment]
@aaemnnosttv Can you review the PR again now that the merge conflicts have been resolved. thanks!

File: assets/js/modules/analytics/util/index.js (hunk 1 above)
Comment (aaemnnosttv) [reply assets/js/modules/analytics/util/index.js:627]: That's good! 😄

File: .storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Storybook main config.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
```
Comment (aaemnnosttv) [inline .storybook/main.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2021 Google LLC
```

**Review by aaemnnosttv — APPROVED** [review]
LGTM 👍

**Review by tofumatt — APPROVED** [review]
I had a quick look over this too and it looks good. Best to merge sooner-than-later to prevent having to keep fixing conflicts so we can start on migrating/defining our approach for new stories 🙂 

Awesome work here @asvinb 👍🏻

**Pull request merged**